### ISSUE TITLE
Intrinsics: unify verify_args with the error checking in create_*

### ISSUE THREAD

**Issue by certik** [lfortran/lfortran#1968]
Currently the logic in `verify_args` is almost exactly identical to the error checking in the `create_*` functions. Let's figure out how to unify it. Some ideas:

* There will be just one verify/check function, that checks arguments and reports all errors into `diagnostics` and returns true/false (no errors vs fail)
* This function will be called by ASR verify, and if it fails, the ASR verify fails
* The same function will be called in `create_*` and if it fails, the AST->ASR will report an error message
* It looks like the `create_*` function has to accept a `diagnostics` variable, and then communicate a flag ok/fail to the caller.

**Comment by czgdp1807** [comment]
Makes sense to me. I have felt the some when implementing intrinsics. If someone can do it then it would be great.
